src/cpp/ext/otel/otel_plugin.h

```diff
@@ -390,6 +419,56 @@ class OpenTelemetryPlugin : public grpc_core::StatsPlugin {
     return plugin_options_;
   }
 
+  template <typename ValueType>
+  struct GaugeState {
+    grpc_core::GlobalInstrumentsRegistry::InstrumentID id;
+    opentelemetry::nostd::shared_ptr<
+        opentelemetry::metrics::ObservableInstrument>
+        instrument;
+    // Per-instrument mutex.
+    grpc_core::Mutex mu;
+    bool ot_callback_registered ABSL_GUARDED_BY(mu) = false;
+    // Cache, it's possible to set values for multiple sets of labels at the
+    // same time.
+    absl::flat_hash_map<std::pair<std::vector<absl::string_view>,
+                                  std::vector<absl::string_view>>,
+                        ValueType>
+        cache ABSL_GUARDED_BY(mu);
+    OpenTelemetryPlugin* ot_plugin;
+  };
+  template <typename T>
+  static void GaugeCallback(opentelemetry::metrics::ObserverResult result,
+                            void* arg);
+
+  template <typename ValueType>
+  struct CallbackGaugeState {
+    // It's possible to set values for multiple sets of labels at the same time
+    // in a single callback.
+    using Cache = absl::flat_hash_map<std::pair<std::vector<absl::string_view>,
```

Comment:
can you document that the first vector is the required set of labels and the second vector is the optional labels?